src/js/controllers/chatController.js

```diff
@@ -62,51 +62,46 @@ module.exports = ViewController.extend({
     },
 
     sendMessage: function(text) {
-        return $.Deferred().resolve().then(function() {
-            var promise = $.Deferred();
-
-            if (this.conversation.isNew()) {
-                this.conversation = this.collection.create(this.conversation, {
-                    success: promise.resolve,
-                    error: promise.reject
-                });
-            } else {
-                promise.resolve(this.conversation);
-            }
-
-            return promise;
-        }.bind(this))
```

Comment:
Good question. I have strong feelings against using `var self = this`, but in this case it would mean adding `.bind(this)` on all Promise functions and it looked sooooo ugly and unreadable that I settled for a small `var self = this` at the top of the chain.

I can't wait to have ES6 support where fat arrows function will take care of that!

I started this PR by adding babel in the build process but it added too much weight to our lib.